libsql/src/hrana/mod.rs

```diff
@@ -123,9 +123,21 @@ impl<T> Statement<T>
 where
     T: HttpSend,
 {
-    pub(crate) fn new(stream: HranaStream<T>, sql: String, want_rows: bool) -> Self {
-        let inner = Stmt::new(sql, want_rows);
-        Statement { stream, inner }
+    pub(crate) fn new(stream: HranaStream<T>, sql: String, want_rows: bool) -> crate::Result<Self> {
+        // in SQLite when a multiple statements are glued together into one string, only the first one is
+        // executed and then a handle to continue execution is returned. However Hrana API doesn't allow
+        // passing multi-statement strings, so we just pick first one.
+        let mut parse = crate::parser::Statement::parse(&sql);
+        match parse.next() {
+            None => Err(crate::Error::Misuse(
+                "no SQL statement provided".to_string(),
+            )),
+            Some(Err(e)) => Err(e.into()),
+            Some(Ok(stmt)) => {
+                let inner = Stmt::new(stmt.stmt, want_rows);
+                Ok(Statement { stream, inner })
+            }
+        }
```

Comment:
I think the issue is with the API for execute_batch. It shouldn't take a coma separated list of statements, but something that implement something like `IntoIterator<Item = Into<Query>>`. Query could be implemented for `(String, Params)`, so that parameters can be bound to each statement in the batch

Reply:
that's what hrana does. it's libsql client that do not follow those semantics. Hrana exposes batching mechanism that sits on top of sqlite semantics, with sole purpose to reduce network roundtrips

Reply:
I don't think so. Sending StreamExecuteReq  with single statement `select 1; select 2` returns an error instead of executing the first statement. Same goes for BatchReq with single batch step having multiple SQLite statements inside of it.

Reply:
yes, we decided to return an error, rather than failing silently on this one. We just shouldn't support multiple statement per query-string

Reply:
the added value is strictly 0

Reply:
it's either we have an execute api like https://www.sqlite.org/c3ref/exec.html, but that doesn't support parameters, or we support parameters and not multi-statement strings

Reply:
but honnestly this api maybe makes sense in C, but in higher level languages, passing an array of statements is very easy

Reply:
@MarinPostma The issue is not with C but other bindings also require the ability to execute multiple statements. For example, Python has this https://docs.python.org/3/library/sqlite3.html#sqlite3.Cursor.executescript which we need to support.